### ISSUE TITLE
"Loading the suggested companies in the wizard failed" and unable to report it

### ISSUE THREAD

**Issue by rugk** [datenanfragen/website#889]
## STR

Just visited the website https://www.datenanfragen.de/ on some device where I have not visited it for some time.

### What happens

I got this window.

![grafik](https://user-images.githubusercontent.com/11966684/168844375-6fe6e789-acbd-414b-a093-94b82f640e5d.png)

I clicked on "Report via GitHub" and it seem to reload the site or whatever and I got the same error.
Then tried "Report via mail" followed by "Report via GitHub" again, nothing does anything at all anymore now… :upside_down_face: 
Then I opened the console manually and now going to copy it manually...

## What should happen

0. Obviously, first of all, no error.
1. It seems to try to load the suggested companies just when visiting the website, IMHO this is not needed yet and may be loaded l,ater (not on the homepage).
2. There are also some "link-preload" that may be corrected.
3. The buttons in that error dialog should actually work/do sth. I don't know what, but they should do sth.
   I remember on first load the GitHub-button actualyl had a link going to GitHub, but now on that second load it seems none of the buttons actually has _any_ href attribute:
   ![grafik](https://user-images.githubusercontent.com/11966684/168846748-2d307607-97c0-46b2-a6c0-04e3c2d137cb.png)
   
   ```html
        <button class="button-unstyled close-button icon-close" title="undefined"></button>

        
        <p>Es ist leider ein unerwarteter Fehler aufgetreten. Wir wären Dir sehr dankbar, wenn Du Dir die Zeit nehmen würdest, uns eine Fehlermeldung über GitHub oder per E-Mail zu senden, sodass wir das Problem diagnostizieren können. Vielen Dank schon einmal!</p>
        <p>Für weitere Informationen dazu, welche Daten Fehlermeldungen enthalten und wie wir diese verarbeiten, schau Dir bitte unsere <a href="/privacy#fehlermeldungen" target="_blank">Datenschutzerklärung</a> an.</p>
        <a id="error-modal-github-link" class="button button-small" style="margin-right: 10px;" target="_blank" rel="noopener noreferrer">
            Über GitHub melden
        </a>
        <a id="error-modal-email-link" class="button button-small">
            Per E-Mail melden
        </a>
    ```

## Console log

![grafik](https://user-images.githubusercontent.com/11966684/168845017-46fa06e5-b41f-40b7-b6ea-d0e978ae758b.png)

> 17:03:12.360 [CanvasBlocker] invalid content script order: require not defined at https://www.datenanfragen.de/ 3949cbd6-1079-45c5-9254-bb2e97511019:17:16
17:03:12.622 An unexpected error occurred: e[5] is undefined error-handler.bundle.gen.js:18
17:03:12.622
error { target: Window, isTrusted: true, message: "TypeError: e[5] is undefined", filename: "https://www.datenanfragen.de/js/commons.bundle.gen.js", lineno: 25, colno: 5569, error: TypeError, srcElement: Window, eventPhase: 0, bubbles: true, … }
error-handler.bundle.gen.js:18
17:03:12.622 Please submit reports via GitHub (https://github.com/datenanfragen/website/issues) or email (dev@datenanfragen.de) and copy the debug details below. error-handler.bundle.gen.js:18
17:03:12.625 Uncaught TypeError: e[5] is undefined
    g commons.bundle.gen.js:25
    c commons.bundle.gen.js:25
    112 general.bundle.gen.js:18
    112 general.bundle.gen.js:18
    a runtime.gen.js:17
    t runtime.gen.js:17
    r runtime.gen.js:17
    <anonymous> general.bundle.gen.js:18
commons.bundle.gen.js:25
17:03:12.627 Deprecation warning: timeoutSeconds is now renamed to connectionTimeoutSeconds vendors.bundle.gen.js:18
17:03:12.796 An unexpected error occurred: Loading the suggested companies in the wizard failed. error-handler.bundle.gen.js:18
17:03:12.797
error { target: Window, isTrusted: true, message: "Error: Not Found", filename: "https://www.datenanfragen.de/js/home.bundle.gen.js", lineno: 18, colno: 5463, error: Error, srcElement: Window, eventPhase: 0, bubbles: true, … }
error-handler.bundle.gen.js:18
17:03:12.797 Please submit reports via GitHub (https://github.com/datenanfragen/website/issues) or email (dev@datenanfragen.de) and copy the debug details below. error-handler.bundle.gen.js:18
17:03:12.805 Uncaught Error: Not Found
    e home.bundle.gen.js:18
    s home.bundle.gen.js:18
    _invoke home.bundle.gen.js:18
    m home.bundle.gen.js:18
    n home.bundle.gen.js:18
    c home.bundle.gen.js:18
home.bundle.gen.js:18
17:03:22.325 Die Ressource unter "https://static.dacdn.de/fonts/fira-go/FiraGO-1001-Regular.woff2" wurde vorab geladen ("link-preload"), aber nicht innerhalb der ersten Sekunden verwendet. Es wird eine Überprüfung empfohlen, ob alle preload-Attribute korrekt gesetzt sind. www.datenanfragen.de
17:03:22.325 Die Ressource unter "https://static.dacdn.de/fonts/fira-go/FiraGO-1001-SemiBold.woff2" wurde vorab geladen ("link-preload"), aber nicht innerhalb der ersten Sekunden verwendet. Es wird eine Überprüfung empfohlen, ob alle preload-Attribute korrekt gesetzt sind. www.datenanfragen.de
17:03:22.325 Die Ressource unter "https://static.dacdn.de/fonts/fira-go/FiraGO-1001-Italic.woff2" wurde vorab geladen ("link-preload"), aber nicht innerhalb der ersten Sekunden verwendet. Es wird eine Überprüfung empfohlen, ob alle preload-Attribute korrekt gesetzt sind. www.datenanfragen.de
17:03:22.325 Die Ressource unter "https://static.dacdn.de/fonts/spartan-mb/SpartanMB-1005-ExtraBold.woff" wurde vorab geladen ("link-preload"), aber nicht innerhalb der ersten Sekunden verwendet. Es wird eine Überprüfung empfohlen, ob alle preload-Attribute korrekt gesetzt sind. www.datenanfragen.de

The error objects mentioned (copied and prettified):
```json
{
  "message": "TypeError: e[5] is undefined",
  "type": "error",
  "colno": 5569,
  "filename": "https://www.datenanfragen.de/js/commons.bundle.gen.js",
  "lineno": 25,
  "error": {
    "message": "e[5] is undefined",
    "name": "TypeError",
    "stack": "g@https://www.datenanfragen.de/js/commons.bundle.gen.js:25:5569\nc@https://www.datenanfragen.de/js/commons.bundle.gen.js:25:17483\n112/<@https://www.datenanfragen.de/js/general.bundle.gen.js:18:1494\n112@https://www.datenanfragen.de/js/general.bundle.gen.js:18:2287\na@https://www.datenanfragen.de/js/runtime.gen.js:17:564\nt@https://www.datenanfragen.de/js/runtime.gen.js:17:433\nr@https://www.datenanfragen.de/js/runtime.gen.js:17:296\n@https://www.datenanfragen.de/js/general.bundle.gen.js:18:47\n"
  },
  "defaultPrevented": false,
  "eventPhase": 2,
  "isTrusted": true,
  "returnValue": true,
  "code_version": "1.0.0",
  "user_agent": "Mozilla/5.0 (X11; Linux x86_64; rv:100.0) Gecko/20100101 Firefox/100.0",
  "url": {
    "href": "https://www.datenanfragen.de/",
    "origin": "https://www.datenanfragen.de",
    "protocol": "https:",
    "host": "www.datenanfragen.de",
    "hostname": "www.datenanfragen.de",
    "port": "",
    "pathname": "/",
    "search": "",
    "hash": ""
  }
}
```

> 17:03:12.625 Uncaught TypeError: e[5] is undefined
    g commons.bundle.gen.js:25
    c commons.bundle.gen.js:25
    112 general.bundle.gen.js:18
    112 general.bundle.gen.js:18
    a runtime.gen.js:17
    t runtime.gen.js:17
    r runtime.gen.js:17
    <anonymous> general.bundle.gen.js:18
commons.bundle.gen.js:25
    g commons.bundle.gen.js:25
    c commons.bundle.gen.js:25
    112 general.bundle.gen.js:18
    112 general.bundle.gen.js:18
    a runtime.gen.js:17
    t runtime.gen.js:17
    r runtime.gen.js:17
    <anonym> general.bundle.gen.js:18

And this one:
```json
{
  "message": "Error: Not Found",
  "type": "error",
  "colno": 5463,
  "filename": "https://www.datenanfragen.de/js/home.bundle.gen.js",
  "lineno": 18,
  "error": {
    "message": "Not Found",
    "description": "Loading the suggested companies in the wizard failed.",
    "name": "Error",
    "stack": "102/p/</e/<@https://www.datenanfragen.de/js/home.bundle.gen.js:18:5463\ns@https://www.datenanfragen.de/js/home.bundle.gen.js:18:14347\n58/u/a._invoke@https://www.datenanfragen.de/js/home.bundle.gen.js:18:14101\n58/m/</<@https://www.datenanfragen.de/js/home.bundle.gen.js:18:14706\nn@https://www.datenanfragen.de/js/home.bundle.gen.js:18:11003\nc@https://www.datenanfragen.de/js/home.bundle.gen.js:18:11215\n",
    "context": {
      "suggested_companies_url": "https://www.datenanfragen.de/db/suggested-companies/undefined_wizard.json"
    }
  },
  "defaultPrevented": false,
  "eventPhase": 2,
  "isTrusted": true,
  "returnValue": true,
  "code_version": "1.0.0",
  "user_agent": "Mozilla/5.0 (X11; Linux x86_64; rv:100.0) Gecko/20100101 Firefox/100.0",
  "url": {
    "href": "https://www.datenanfragen.de/",
    "origin": "https://www.datenanfragen.de",
    "protocol": "https:",
    "host": "www.datenanfragen.de",
    "hostname": "www.datenanfragen.de",
    "port": "",
    "pathname": "/",
    "search": "",
    "hash": ""
  }
}
```

> 17:03:12.805 Uncaught Error: Not Found
    e home.bundle.gen.js:18
    s home.bundle.gen.js:18
    _invoke home.bundle.gen.js:18
    m home.bundle.gen.js:18
    n home.bundle.gen.js:18
    c home.bundle.gen.js:18
home.bundle.gen.js:18
    e home.bundle.gen.js:18
    s home.bundle.gen.js:18
    _invoke home.bundle.gen.js:18
    m home.bundle.gen.js:18
    n home.bundle.gen.js:18
    c home.bundle.gen.js:18

## System

Firefox 100.0
Mozilla/5.0 (X11; Linux x86_64; rv:100.0) Gecko/20100101 Firefox/100.0
Linux 5.17.6-300.fc36.x86_64 #1 SMP PREEMPT Mon May 9 15:47:11 UTC 2022
Fedora 36 Workstation

I have installed many blocking extension, but I don't see anyone that did something/could interfere. A new/rare one is possibly [CookieBlock](https://github.com/dibollinger/CookieBlock).

**Comment by rugk** [comment]
Ah, it actually seems to be reproducible.

It tries to load some strange URL and fails there:
![grafik](https://user-images.githubusercontent.com/11966684/168847277-ba34ec84-5124-4ab9-99d0-bea9901e5c96.png)

```js
await fetch("https://www.datenanfragen.de/db/suggested-companies/undefined_wizard.json", {
    "credentials": "omit",
    "headers": {
        "User-Agent": "Mozilla/5.0 (X11; Linux x86_64; rv:100.0) Gecko/20100101 Firefox/100.0",
        "Accept": "*/*",
        "Accept-Language": "de,en-US;q=0.7,en;q=0.3",
        "Sec-Fetch-Dest": "empty",
        "Sec-Fetch-Mode": "cors",
        "Sec-Fetch-Site": "same-origin",
        "Sec-GPC": "1"
    },
    "method": "GET",
    "mode": "cors"
});
```

```shell
curl 'https://www.datenanfragen.de/db/suggested-companies/undefined_wizard.json' -H 'User-Agent: Mozilla/5.0 (X11; Linux x86_64; rv:100.0) Gecko/20100101 Firefox/100.0' -H 'Accept: */*' -H 'Accept-Language: de,en-US;q=0.7,en;q=0.3' -H 'Accept-Encoding: gzip, deflate, br' -H 'DNT: 1' -H 'Connection: keep-alive' -H 'Sec-Fetch-Dest: empty' -H 'Sec-Fetch-Mode: cors' -H 'Sec-Fetch-Site: same-origin' -H 'Sec-GPC: 1' -H 'TE: trailers'
```

**Comment by rugk** [comment]
## Local storage

Looks okayish:
![grafik](https://user-images.githubusercontent.com/11966684/168847709-9097e867-0dac-4aec-9b25-42f19e2fe43c.png)

Also the other data is still there and looks valid, even the wizard company list is filled:
![grafik](https://user-images.githubusercontent.com/11966684/168847926-842e3722-62c4-4099-b949-cc289f802da7.png)

**Comment by rugk** [comment]
Maybe the issue/error that leads to your error dialog not working is that the dialog pops up twice... I checked it with breaking on exceptions.
I.e. first the first error happens, then the dialog still looks fine…

![grafik](https://user-images.githubusercontent.com/11966684/168848830-aa6dbfbf-7da3-49fc-a050-e2cd963beaff.png)

…but when I continue and the dialog pops up again due to (subsequent fault/error)… it's empty.

Or well... maybe the second erorr itself has problems to be reported as such.

**Comment by rugk** [comment]
BTW I can also not close the modal… :upside_down_face:

**Comment by rugk** [comment]
BTW I can also reproduce it on another device, so definitively an error here…

**Comment by baltpeter** [comment]
> BTW I can also reproduce it on another device, so definitively an error here…

Oh, that's worrying. :| I'll have to do some more testing myself then.

**Comment by baltpeter** [comment]
> Maybe the issue/error that leads to your error dialog not working is that the dialog pops up twice... I checked it with breaking on exceptions. I.e. first the first error happens, then the dialog still looks fine…
> 
> ![grafik](https://user-images.githubusercontent.com/11966684/168848830-aa6dbfbf-7da3-49fc-a050-e2cd963beaff.png)
> 
> …but when I continue and the dialog pops up again due to (subsequent fault/error)… it's empty.
> 
> Or well... maybe the second erorr itself has problems to be reported as such.

You're right. I am able to reproduce the error modal problem whenever a second error is thrown. That's good to know.

**Comment by rugk** [comment]
Well oaky, now the descriptive error is not catched anymore, but only that other strange error ("e[5] is undefined")... I'm not sure this is better...

![grafik](https://user-images.githubusercontent.com/11966684/168870079-bbcfd3ed-4a65-4932-bf56-e81f86a8e975.png)

**Comment by zner0L** [comment]
Do you still see the error if you clear your localStorage?

**Comment by rugk** [comment]
(Closing button works again, at least.)

**Comment by zner0L** [comment]
Ok, I narrowed the error down to `js-cookie` because it is bundled in the `common.bundle.gen.js` in line 25 as the debug info states. I think that one of your extentions (I am suspecting CookieBlock) might interfere here, since we delete the cookie and then try to read an empty one. Although this should still be fixed, can you try disabeling CookieBlock and clearing your localStorage?

**Comment by rugk** [comment]
Tried. Also happens with cookie block "paused":
![grafik](https://user-images.githubusercontent.com/11966684/168872659-5233ea78-15dd-4ad4-aae0-412134606d9b.png)

**Comment by baltpeter** [comment]
@zner0L We could probably even undo eba94e5f1e8f7217ee0423350f85f42a5334c801. Or we could decide to keep it just in case. Not sure.

**Comment by zner0L** [comment]
Well, I wanted that in the first place, so I vote keep.

**Comment by baltpeter** [comment]
I'll leave this open until the fix is confirmed by @rugk.

**Comment by baltpeter** [comment]
Great, thank you!

The `connectionTimeoutSeconds` warning should be fixed by #887.
